### PR TITLE
Mod customization header/panel changes

### PR DESCRIPTION
**Pull request by Fabep**
Removed on click event for the mod customization header, As of #29618 .

Renamed ModCustomisationPanelState, since only mods set the "old" Expanded panel state.

Further suggestions that could be applied to this area:

- Refactor mods that use the ExpandedByMod state to set parameters directly, for example copy attributes from current map when you enable difficulty adjustments.
- Keep colourprovider.light4 on the header when the mod customization panel is expanded. (I attempted to add this, but I am too stupid to understand how to remove it when the hover is lost & the panel is collapsed). 😃

### CODE REVIEW

**Comment by peppy** [comment]
@Fabep any reason this is marked as a draft?

**Comment by bdach** [comment]
I have big suspicions this just straight up breaks mobile btw. Haven't confirmed yet.

**Comment by Fabep** [comment]
@peppy I have no clue 😎 
But as bdach said something could break completely for mobile/touch device since the only change is the removal of the click event.

**Comment by peppy** [comment]
Reading it, I can't see how it would break mobile. But either way let's just undraft it for proper review.

**Review by peppy — DISMISSED** [review]
Haven't tested mobile but seems fine otherwise?

**Comment by bdach** [comment]
Touch was likely working only by way of the touch-to-synthesised-click pathway, which is evidenced further by stuff like this:

https://github.com/ppy/osu/blob/363fc0d5da59164b6f348fa6f02715e7d478b236/osu.Game/Overlays/Mods/ModCustomisationHeader.cs#L117-L121
https://github.com/ppy/osu/blob/363fc0d5da59164b6f348fa6f02715e7d478b236/osu.Game/Overlays/Mods/ModCustomisationHeader.cs#L132-L133

So if my hunch is correct then the panel just won't open on mobile at all.

**Comment by bdach** [comment]
As I suspected, this just broke touch completely. https://github.com/ppy/osu/pull/29626/commits/5211e606b5b0c9919a2675c09a263ec88acb8c0d *should* bring the behaviour more or less to parity with desktop:

https://github.com/user-attachments/assets/aa26ca1f-e0fc-4466-bced-afee93922242

tl;dr: you have to touch the dropdown to open it if it's not forced open by a mod, and touch away from it to close it. Touching the dropdown header when open does nothing which matches desktop.

@peppy requesting re-check to make sure this is the UX you are looking for.

**Review by bdach — DISMISSED** [review]

**Comment by Fabep** [comment]
>     Keep colourprovider.light4 on the header when the mod customization panel is expanded. (I attempted to add this, but I am too stupid to understand how to remove it when the hover is lost & the panel is collapsed). 😃

Personally for UI purposes, I believe this to be the only thing that could be improved upon. The hover color is now redundant due to the fact that the only way to open the mod customization panel is through the hover (or force open by mod). It would be more clear and "cleaner" to bind the color to it being active or not, e.g dark when collapsed and light when expanded. 

I've tried implementing it with the following code snippet inside of the ModCustomisationHeader class.
```cs
protected override void LoadComplete()
{
    ...
    panel.ExpandedState.BindValueChanged(v =>
    {
        IdleColour = v.NewValue switch
        {
            ModCustomisationPanelState.Expanded or
            ModCustomisationPanelState.ExpandedByMod => colourProvider.Light4,
            _ => colourProvider.Dark3,
        };
    }, true);
    ...
}
```

However when I try it functionally it works like the following: 

https://github.com/user-attachments/assets/56720f9a-7265-45b3-a973-b729e35803f6

**Comment by peppy** [comment]
I'm reverting the colour changes. Please make them in a separate PR or something. They look weird and your code formatting is wrong.

**Review by peppy — APPROVED** [review]
